src/Adapter/Product/Update/ProductPricePropertiesFiller.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use PrestaShop\Decimal\DecimalNumber;
+use Product;
+
+/**
+ * Fills Product price properties which needs specific handling for update
+ */
+class ProductPricePropertiesFiller
+{
+    /**
+     * Wraps following properties filling: price, unit_price, unit_price_ratio, wholesale_price
+     * as most of them (price, unit_price, unit_price_ratio) are highly coupled & depends on each other
+     *
+     * @param Product $product
+     * @param DecimalNumber|null $price
+     * @param DecimalNumber|null $unitPrice
+     * @param DecimalNumber|null $wholesalePrice
+     *
+     * @return string[] updatable properties
+     */
+    public function fillWithPrices(Product $product, ?DecimalNumber $price, ?DecimalNumber $unitPrice, ?DecimalNumber $wholesalePrice): array
+    {
+        if (null !== $wholesalePrice) {
+            $product->wholesale_price = (string) $wholesalePrice;
+            $updatableProperties[] = 'wholesale_price';
+        }
+
+        if (null !== $price) {
+            $product->price = (float) (string) $price;
+            $updatableProperties[] = 'price';
+        } else {
+            $price = new DecimalNumber((string) $product->price);
+        }
+
+        $this->fillUnitPriceRatio($product, $price, $unitPrice);
```

Comment:
So price ratio will always be updated no matter the input This could probably be avoided in case both `$price` and `$unitPrice` are null But it's a negligible optimization so it doesn't matter